src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -660,46 +699,81 @@ private static ImmutableArray<(IParameterSymbol parameter, bool before)> GetSibl
+                            return (property, false);
                     }
                 }
             }
 
             // Couldn't find any existing member.  Just return nothing so we can offer to
             // create a member for them.
-            return null;
-        }
+            return default;
 
-        private static bool ContainsMemberAssignment(
-            IBlockOperation? blockStatementOpt, ISymbol member)
-        {
-            if (blockStatementOpt != null)
+            static bool ContainsMemberAssignment(IBlockOperation? blockStatement, ISymbol member)
             {
-                foreach (var statement in blockStatementOpt.Operations)
+                if (blockStatement != null)
                 {
-                    if (IsFieldOrPropertyAssignment(statement, member.ContainingType, out var assignmentExpression) &&
-                        UnwrapImplicitConversion(assignmentExpression.Target) is IMemberReferenceOperation memberReference &&
-                        member.Equals(memberReference.Member))
+                    foreach (var statement in blockStatement.Operations)
                     {
-                        return true;
+                        if (IsFieldOrPropertyAssignment(statement, member.ContainingType, out var assignmentExpression) &&
+                            assignmentExpression.Target.UnwrapImplicitConversion() is IMemberReferenceOperation memberReference &&
+                            member.Equals(memberReference.Member))
+                        {
+                            return true;
+                        }
                     }
                 }
+
+                return false;
             }
 
-            return false;
+            bool IsThrowNotImplementedProperty(IPropertySymbol property)
+            {
+                using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var accessors);
+
+                if (property.GetMethod != null)
+                    accessors.AddIfNotNull(GetAccessorBody(property.GetMethod, cancellationToken));
+
+                if (property.SetMethod != null)
+                    accessors.AddIfNotNull(GetAccessorBody(property.SetMethod, cancellationToken));
+
+                if (accessors.Count == 0)
+                    return false;
+
+                foreach (var group in accessors.GroupBy(node => node.SyntaxTree))
+                {
+                    var semanticModel = compilation.GetSemanticModel(group.Key);
+                    foreach (var accessorBody in accessors)
+                    {
+                        var operation = semanticModel.GetOperation(accessorBody, cancellationToken);
+                        if (operation is null)
+                            return false;
+
+                        if (!operation.IsSingleThrowNotImplementedOperation())
+                            return false;
+                    }
+                }
+
+                return true;
+            }
```

Comment:
this is where we detect and allow a property that contains `throw NotImplementedException`.